File: frontend/src/order/object/after-sales/SubmitAfterSalesRequest.ts
import type {
  AfterSalesRequestType,
  CurrencyCents,
  ReasonText,
} from '@/shared/object/domain/DomainObjects'

export type SubmitAfterSalesRequest = {
  requestType: AfterSalesRequestType
  reason: ReasonText
  expectedCompensationCents?: CurrencyCents
}

File: backend/src/main/scala/domain/order/SubmitAfterSalesRequest.scala
package domain.order

import domain.shared.given

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.*
import domain.shared.*

final case class SubmitAfterSalesRequest(
    requestType: AfterSalesRequestType,
    reason: ReasonText,
    expectedCompensationCents: Option[CurrencyCents],
)
object SubmitAfterSalesRequest:
  given Encoder[SubmitAfterSalesRequest] = deriveEncoder
  given Decoder[SubmitAfterSalesRequest] = deriveDecoder
